grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -834,16 +834,14 @@ class RequisitionService {
 
     void triggerRequisitionStatusTransition(Requisition requisition, User currentUser) {
         // OBPIH-5134 Request approval feature implements additional status transitions for a request
-        // If fulfilling location does not require approval we omit all other status transitions and set it to VERIFYING
         switch(requisition.status) {
+            case RequisitionStatus.VERIFYING:
+                transitionRequisitionStatus(requisition, RequisitionStatus.VERIFYING, EventCode.SUBMITTED, currentUser)
+                requisition.approvalRequired = false
+                break
             case RequisitionStatus.PENDING_APPROVAL:
-                if (requisition.origin.approvalRequired) {
-                    transitionRequisitionStatus(requisition, RequisitionStatus.PENDING_APPROVAL, EventCode.PENDING_APPROVAL, currentUser)
-                    requisition.approvalRequired = true
-                } else {
-                    transitionRequisitionStatus(requisition, RequisitionStatus.VERIFYING, EventCode.SUBMITTED, currentUser)
-                    requisition.approvalRequired = false
-                }
```

Comment:
why have you got rid of those lines?
If you changed on frontend to send `PENDING_APPROVAL` no matter of location supported activities, if a location doesn't support the approvals it would set the status for `PENDING_APPROVAL` not for `VERIFYING`, wouldn't it?

Reply:
yes, good point., this is what I wanted to hear, because I had a feeling I was missing something.
I'll bring back those lines and leave the VERYFING case too.